src/main/java/net/snowflake/ingest/streaming/internal/SnowflakeStreamingIngestChannelInternal.java

```diff
@@ -291,6 +297,19 @@ public CompletableFuture<Void> close() {
                         .map(SnowflakeStreamingIngestChannelInternal::getFullyQualifiedName)
                         .collect(Collectors.toList()));
               }
+              if (this.dropOnClose) {
```

Comment:
should we drop the channel even if it hits the exception above?

Reply:
In theory, I don't see an issue of dropping a closed channel, should this be removed?

Reply:
Also what about when a channel becomes invalid, currently we can't close an invalid channel, but we should allow dropping them?

Reply:
I think dropping after failing to commit would be a surprising behavior. Typically, when `close` fails, the caller is supposed to handle it e.g. by retrying etc. Dropping in this specific case would lose uncommitted data.

Reply:
I think it depends, I can imagine a use case where customer always open a new channel with an unique identifier every time the old channel gets invalidated. I guess we can come back to this if needed in the future, basically right now we won't drop any closed and invalid channels.